lnwallet/channel.go

```diff
@@ -3896,16 +4035,26 @@ func (lc *LightningChannel) validateCommitmentSanity(theirLogCounter,
 		lc.channelState.RemoteChanCfg.ChanReserve,
 	)
 
+	// Calculate the commitment fee to log the information if needed.
+	commitFee := feePerKw.FeeForWeight(commitWeight)
+	commitFeeMsat := lnwire.NewMSatFromSatoshis(commitFee)
+
 	switch {
+	// TODO(ziggie): Remove the InitialBalance check?
```

Comment:
So I thought about the argument in https://github.com/lightningnetwork/lnd/issues/8249  and I think we should include this if we wanna fix the stuck channel and a potential splicing case where the reserve can increase and we need to push funds to the remote. I did not include it in a commit for now because it broke several testcases again, so first waiting on your comments.
```

	ourBalanceDecreasing := ourBalance < ourInitialBalance
	theirBalanceDecreasing := theirBalance < theirInitialBalance

	switch {
	// We do not allow the dipping below the channel reserve if their
	// balance is NOT decreasing which means they are not adding an htlc.
	case !theirBalanceDecreasing && ourBalance < ourReserve:
		lc.log.Debugf("Funds below chan reserve: ourBalance=%v, "+
			"ourReserve=%v, commitFee=%v, feeBuffer=%v "+
			"chan_initiator=%v", ourBalance, ourReserve,
			commitFeeMsat, bufferAmt, lc.channelState.IsInitiator)

		return fmt.Errorf("%w: our balance below chan reserve",
			ErrBelowChanReserve)

	// We do not allow the dipping below their channel reserve if our amount
	// is NOT decreasing which means we are not adding any htlcs.
	case !ourBalanceDecreasing && theirBalance < theirReserve:
		lc.log.Debugf("Funds below chan reserve: theirBalance=%v, "+
			"theirReserve=%v", theirBalance, theirReserve)

		return fmt.Errorf("%w: their balance below chan reserve",
			ErrBelowChanReserve)

	// We allow our balance to dip below the channel reserve in case the
	// balance of our peer is decreasing which means an incoming htlc is
	// added to the commitment state. We carve out this exception to account
	// for stuck channel prior to introducing the fee buffer and in forsight
	// of splicing where increasing the channel capacity can increase the
	// reserve requirement.
	case theirBalanceDecreasing && ourBalance < ourReserve:
		lc.log.Debugf("Funds below chan reserve but our amount is "+
			"increasing: ourInitialBalance=%v, ourBalance=%v, "+
			"ourReserve=%v", ourInitialBalance, ourBalance,
			ourReserve)

	// We allow their balance to dip below the channel reserve in case our
	// balance is decreasing which means we are adding an outgoing htlc to
	// the commitment state. We carve out this exception to account
	// for stuck channel prior to introducing the fee buffer and in forsight
	// of splicing where increasing the channel capacity can increase the
	// reserve requirement.
	case ourBalanceDecreasing && theirBalance < theirReserve:
		lc.log.Debugf("Funds below chan reserve but their amount is "+
			"increasing: theirInitialBalance=%v, ourBalance=%v, "+
			"ourReserve=%v", theirInitialBalance, ourBalance,
			ourReserve)
	}
```

Reply:
agree, will reformulate the TODO a bit, because the `ourInitialBalance ` will come useful so no need to remove it.